### PR TITLE
Signup

### PR DESCRIPTION
**Pull request by olavea**
Look at this comment in issue #91  
https://github.com/raae/pow-app/issues/91#issuecomment-590706626
✌️😺 👍

### CODE REVIEW

**Comment by olavea** [comment]
POW-_userbase_App ID not validApplication ID must be a string -86

**Comment by raae** [comment]
Very good, you know have a signup form. What I would like it the UserForm to be both a signup form and a signin form. So we can delete SignInForm.

The way I would like to use it is
````
<UserForm variant="signup" />
```
and 
````
<UserForm variant="signin" />
```

Continue with what you have make use of the variant prop to decide:
- if it should do signup or sign in action in handle submit
- what text "Sign Up" or "Log in" to show on the button
- what helper text to show below the button

File: src/components/UserForm.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  },
+  alert: {
+    margin: theme.spacing(3, 0, 1),
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 3),
+  },
+}))
+
+const UserForm = () => {
```
Comment (raae) [inline src/components/UserForm.js]: You must get hold of the prop here.
```
const UserForm = ({variant}) => {
```

File: src/components/UserForm.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      return {
+        ...prevState,
+        [name]: value,
+      }
+    })
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    signUp(state).then(() => {
```
Comment (raae) [inline src/components/UserForm.js]: Then use variant prop here 

```
if(variant === "signup") {
// Do signup stuff
} else {
// If no variant given lets asume we want signIn 
// Do SignIn stuff
}
```

File: src/components/UserForm.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        )}
+        <Button
+          className={classes.submit}
+          disabled={isPending}
+          type="submit"
+          fullWidth
+          variant="contained"
+          color="primary"
+        >
+          Sign Up
```
Comment (raae) [inline src/components/UserForm.js]: And use it here using inline conditional logic
```
// If variant is equal to "signup" then return "Sign Up" else return "Log In"
{variant === "signup ? "Sign Up" : "Log In" }
```

File: src/components/UserForm.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        </Button>
+        <Typography variant="body2" align="right">
+          Already have an account?&nbsp;
```
Comment (raae) [inline src/components/UserForm.js]: Do same as with button, but returning the whole Typography component for each case.

**Comment by olavea** [comment]
😺 👍 
Take another look now 
@raae

File: src/components/UserForm.js (hunk 3 above)
Comment (olavea) [reply src/components/UserForm.js]: I read "And use it here using inline conditional MAGIC" 😸

**Comment by raae** [comment]
I think i managed to approve and you can go ahead and merge!

**Pull request merged**